### PR TITLE
Issue1202 smooth heavisidev2

### LINKED ISSUE #1202: SmoothHeaviside
Twice continuously differentiable heaviside function: y = max(0, min(1, x^3*(10+x*(-15+6*x))));

### PR DESCRIPTION
**Pull request by Mathadon**
closes #1202

### CODE REVIEW

File: IBPSA/Utilities/Math/Functions/smoothHeaviside.mo [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
+ y := max(0, min(1, x^3*(10+x*(-15+6*x))));
+
+ annotation (smoothOrder = 2,
  Documentation(info="<html>
 <p>
-Once Lipschitz continuously differentiable approximation to the
-<code>Heaviside(.,.)</code> function.
+Twice continuously differentiable approximation to the
+<code>Heaviside(.,.)</code> function.<br/>
+Function is derived from a quintic polynomal going through (0,0) and (1,1),
```
Comment (Mathadon) [inline IBPSA/Utilities/Math/Functions/smoothHeaviside.mo]:
```suggestion
Function is derived from a quintic polynomial going through (0,0) and (1,1), 
```

File: IBPSA/Utilities/Math/Functions/smoothHeaviside.mo [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
+<code>Heaviside(.,.)</code> function.<br/>
+Function is derived from a quintic polynomial going through (0,0) and (1,1), 
+with zero first and second order derivatives at those points.<br/>
 See Example <a href=\"modelica://IBPSA.Utilities.Math.Examples.SmoothHeaviside\">
 IBPSA.Utilities.Math.Examples.SmoothHeaviside</a>.
 </p>
 </html>", revisions="<html>
 <ul>
 <li>
+Setpember 13, 2019, by Kristoff Six:<br/>
```
Comment (Mathadon) [inline IBPSA/Utilities/Math/Functions/smoothHeaviside.mo]:
```suggestion
September 13, 2019, by Kristoff Six:<br/>
```

**Comment by Mathadon** [comment]
working on unit tests

**Comment by Mathadon** [comment]
I bugged out my dymola installation when upgrading to 2020 so need some time before I can update this PR.

**Comment by Mathadon** [comment]
@mwetter can you review this?

**Comment by mwetter** [comment]
@Mathadon : Is it twice *Lipschitz* differentiable (which I added to the documentation)?

**Comment by Mathadon** [comment]
I looked up what that means exactly. I don't have the mathematical background to judge, but since it's a polynomial and polynomials seems to be Lipschitz continuous for x not equal to infinity I'd say yes.

**Comment by Mathadon** [comment]
Thanks for fixing the documentation. I overlooked it.

**Comment by mwetter** [comment]
@Mathadon : The function is twice Lipschitz continuously differentiable, see https://github.com/ibpsa/modelica-ibpsa/pull/1204/commits/c4702464635033a03b2b72b8bebf4575fe35ee33
For functions that are used in IbpsaMPC, it would be good to add such tests. They also guard against someone changing it, and in this case, it showed that a `smooth(2, ...)` was missing, causing Dymola to issue an error in the pedantic mode.

**Pull request merged**